### PR TITLE
git-my: init at 1.1.2

### PR DESCRIPTION
**Pull request by bb010g**
Introduces [git-my](https://github.com/davidosomething/git-my) into gitAndTools.

Also takes the luxury beforehand to strongly normalize gitAndTools's naming and sorting. The only package referenced elsewhere in Nixpkgs, git-top, has had its reference updated to not point at the new alias. These are done in separate commits to keep things clean.

#### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)

  (Which of these are we supposed to post?)
  - 615_240 (derivation)
  - 29_474_872 (output)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/71378)

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/default.nix [hunk 1]
```diff
@@ -147,39 +149,43 @@ let
...
   svn2git = callPackage ./svn2git {
     git = gitSVN;
   };
 
-  svn-all-fast-export = libsForQt5.callPackage ./svn-all-fast-export { };
-
   tig = callPackage ./tig { };
 
-  topGit = callPackage ./topgit { };
+  top-git = callPackage ./topgit { };
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/default.nix]: although these names should have never been merged in the first place, renaming them can be a breaking change for people who have created workflows around them. If you're going to change them, I would at least do it in another PR where we aren't conflating the addition of git-my and the issue of naming

@worldofpeace thoughts?

Comment (worldofpeace) [reply pkgs/applications/version-management/git-and-tools/default.nix]: It seems @bb010g established aliases so if someone relies on them the alias will make it backwards compatible.
Though it would be better in a PR, but I believe no one has issues with hyphenated attribute names anymore.

Comment (jonringer) [reply pkgs/applications/version-management/git-and-tools/default.nix]: oh, i skipped over the aliases at the bottom, LGTM now

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "0jji5zw25jygj7g4f6f3k0p0s9g37r8iad8pa0s67cxbq2v4sc0v";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    install -Dm755 -t "$out"/bin ./git-my
+  '';
+
+  meta = let inherit (stdenv) lib; in {
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/git-my/default.nix]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    description =
+      "List remote branches if they're merged and/or available locally";
+    homepage = https://github.com/davidosomething/git-my;
+    license = lib.licenses.free;
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/git-my/default.nix]:
```suggestion
    license = licenses.free;
```

Comment (jonringer) [reply pkgs/applications/version-management/git-and-tools/git-my/default.nix]: added an issue for him to add a proper license https://github.com/davidosomething/git-my/issues/3

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    maintainers = let m = lib.maintainers; in [ m.bb010g ];
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/git-my/default.nix]:
```suggestion
    maintainers = with maintainers; [ bb010g ];
```

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    platforms = lib.platforms.all;
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/git-my/default.nix]:
```suggestion
    platforms = platforms.all;
```

**Review by jonringer — COMMENTED** [review]
some small corrections, otherwise LGTM

thank you for organizing the gitAndTools package scope :)

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix (hunk 2 above)
Comment (bb010g) [reply pkgs/applications/version-management/git-and-tools/git-my/default.nix]: For these comments, I have the impression that Nix was moving away from `with` syntax from various conversations here? [The papercut thread discusses it masking the source of variables without evaulation](https://discourse.nixos.org/t/the-papercut-thread-post-your-small-annoyances-confusions-here/3100/6?u=bb010g), NixOS/nix#3042 dislikes it from a gradual typing perspective, and NixOS/nix#1361's discussion includes multiple people in favor of deprivation. I've been avoiding it in my new expressions since learning about it.

Are there places where it's more or less expected in Nixpkgs? (E.g. in a derivation's `meta` attribute vs. a new chunk of significant abstraction.)

Comment (worldofpeace) [reply pkgs/applications/version-management/git-and-tools/git-my/default.nix]: @bb010g It's likely because probably 90% of the expressions for packages in nixpkgs look like this.
For more complicated code you should wary of https://discourse.nixos.org/t/the-papercut-thread-post-your-small-annoyances-confusions-here/3100/6?u=bb010g. This is isolated to `meta` so it's fine.

**Comment by bb010g** [comment]
Ok, reviews incorporated.

**Comment by Lassulus** [comment]
needs a rebase

**Review by jonringer — COMMENTED** [review]
apparently forgot to submit a review, sorry if this has already been addressed

File: pkgs/applications/version-management/git-and-tools/git-my/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "git-my";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "davidosomething";
+    repo = "git-my";
+    rev = version;
+    sha256 = "0jji5zw25jygj7g4f6f3k0p0s9g37r8iad8pa0s67cxbq2v4sc0v";
+  };
+
```
[the remark this answers is not in the document]
Comment (bb010g) [reply pkgs/applications/version-management/git-and-tools/git-my/default.nix]: The other Git & tools scripts don't seem to do this, though?

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/applications/version-management/git-and-tools/git-my/default.nix]: that's fair, i guess it doesn't really make sense to use this without git installed. I'll resolve this

**Comment by jonringer** [comment]
still needs a rebase, i dismissed my comments

**Comment by jonringer** [comment]
still needs a rebase

**Comment by bb010g** [comment]
Rebsed.

**Comment by jonringer** [comment]
hmm, do you mind separating out the `git-my: init at 1.1.2` into a separate PR (that looks read). And make this into the name refactor (more controversial).

I don't want to step on the toes of those maintainers who are having their packages forcefully edited without their knowledge.

**Comment by bb010g** [comment]
`darcs-to-git` doesn't have a maintainer & `hubUnstable` is currently `throw "use gitAndTools.hub instead"`.

`fast-export` is @koral, `git-remote-gcrypt` is @ellis & @montag451, and `top-git` is @MarcWeber.

If they're fine with the renames & aliasing, then this shouldn't need to split?

File: pkgs/applications/version-management/git-and-tools/default.nix [hunk 7]
```diff
@@ -167,43 +169,47 @@ let
     inherit (darwin) Security;
   };
 
-  hubUnstable = throw "use gitAndTools.hub instead";
+  hub-unstable = throw "use gitAndTools.hub instead";
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/default.nix]: This has been here for 2 years, we can remove it
```suggestion
```

**Comment by worldofpeace** [comment]
@bb010g I cool merging this after the previous comment is applied.
@jonringer It's good to look out for what other maintainers think about their packages, but this is more of a thing the committers (curators) of the codebase can decide on.

File: pkgs/applications/version-management/git-and-tools/default.nix [hunk 8]
```diff
@@ -167,43 +169,46 @@ let
...
 
   ydiff = pkgs.python3.pkgs.toPythonApplication pkgs.python3.pkgs.ydiff;
 
 } // lib.optionalAttrs (config.allowAliases or true) (with self; {
   # aliases
+  darcsToGit = darcs-to-git;
   gitAnnex = git-annex;
+  gitFastExport = git-fast-export;
+  gitRemoteGcrypt = git-remote-gcrypt;
+  hubUnstable = hub-unstable;
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/default.nix]: oops, forgot to suggest this one
```suggestion
```

**Comment by bb010g** [comment]
(Huh, GitHub is ordering hub-unstable as the last commit in both the conversations & commits views, even though it's topologically before all the other commits, because of its (severely?) newer author date.)

**Comment by worldofpeace** [comment]
> (Huh, GitHub is ordering hub-unstable as the last commit in both the conversations & commits views, even though it's topologically before all the other commits, because of its (severely?) newer author date.)

I've viewed your stuff locally :+1:
